tests/Core/Content/Product/SalesChannel/Listing/ProductListingFeaturesSubscriberTest.php

```diff
@@ -1043,16 +1053,16 @@ public function testCriteriaLimitIsSetForPagination(
             $expectedAssign['source'] = null;
         }
         $expectedAssign['title'] = null;
+        $expectedAssign['limit'] = $expectedLimit === Pagination::DEFAULT_LIMIT ? null : $expectedLimit;
 
         $criteriaMock = $this->getMockBuilder(Criteria::class)->disableOriginalConstructor()->getMock();
         $invokeCountAssign = $isNavigationRequest ? $this->never() : $this->once();
         $invokeCountOffset = $isNavigationRequest ? $this->never() : $this->exactly(3);
-        $invokeCountLimit = $isNavigationRequest ? $this->once() : $this->exactly(3);
         $criteriaMock->expects($invokeCountAssign)->method('assign')->with($expectedAssign);
         $criteriaMock->expects($invokeCountOffset)->method('getOffset')->willReturn(0);
         // Add this check to ensure that if no custom limit is provided, it uses the default limit
         if ($expectedLimit !== Pagination::DEFAULT_LIMIT) {
-            $criteriaMock->expects($invokeCountLimit)->method('getLimit')->willReturn($expectedLimit);
+            $criteriaMock->expects($this->any())->method('getLimit')->willReturn($expectedLimit);
```

Comment:
From the logic above, I don't understand why this is not called when default limit of 24 is used, please explain.

Reply:
Ye doesn't make sense. I changed it appropiately.